src/Blocks/components/form/assets/form.js

```diff
@@ -1714,17 +1734,23 @@ export class Form {
 
 		this.utils.setOnUserChangeInput(input);
 
-		if (
-			(this.state.getStateConfigIsAdmin() && this.state.getStateElementIsSingleSubmit(name, formId)) ||
-			this.state.getStateFormConfigUseSingleSubmit(formId)
-		) {
+		// Used only for admin single submit.
+		if (this.state.getStateConfigIsAdmin() && this.state.getStateElementIsSingleSubmit(name, formId)) {
 			debounce(
 				this.formSubmit(
 					formId, {
 						[this.FILTER_USE_ONLY_FIELDS]: [name]
 					}
 			), 100);
 		}
+
+		// Used only on frontend for single submit.
+		if (!this.state.getStateConfigIsAdmin() && this.state.getStateFormConfigUseSingleSubmit(formId)) {
+			debounce(
+				this.formSubmit(
+					formId
+			), 100);
```

Comment:
```suggestion
			debounce(this.formSubmit(formId), 100);
```
Change others as well pls, it's hard to read when strung over 4 lines